templates/results.html

```diff
@@ -11,126 +11,124 @@
+<body class="search_results_page">
     <xi:include href="menu.html" />
-    <div class="container" >
-    <div id="content" class="page_content" i18n:comment="On the page of results of a search.">
-
-	<h1><span class="icon icon_${os.title_icon}" />${os.title}</h1>
-
-      <div class="body">
-	<div>
-	  <ul class="results">
-	    <py:for each="n, e in enumerate (os.entries)">
-
-	      <py:if test="e.header and old_header != e.header">
-		<li class="h2">
-		  <h2 class="padded">${e.header}</h2>
-		</li>
-		<?python old_header = e.header ?>
-	      </py:if>
-
-	      <py:if test="isinstance (e, bs.Cat) and e.rel not in ('start', )">
-		<py:choose test="">
-		  <py:when test="e.rel == '__statusline__'">
-		    <li class="statusline">
-		      <div class="padded">
-			<span>${e.title}</span> ${prev_next_links ()}
-		      </div>
-		    </li>
-		    <?python os.status_line = e.title ?>
-		  </py:when>
-
-		  <py:otherwise>
-		    <li class="${e.class_}">
-		      <a class="link" href="${e.url}" accesskey="${str (n % 10)}">
-			<span class="cell leftcell${' with-cover' if e.thumb_url else ' without-cover'}">
-			  <span py:if="not e.thumb_url" class="icon-wrapper">
-			    <span class="icon icon_${e.icon}" />
-			  </span>
-			  <img py:if="e.thumb_url"
-			       class="cover-thumb" src="${e.thumb_url}" alt="" />
-			</span>
-			<span class="cell content">
-			  <span py:if="e.title" class="title">${e.title}</span>
-			  <span py:if="e.subtitle" class="subtitle">${e.subtitle}</span>
-			  <span py:if="e.extra" class="extra">${e.extra}</span>
-			</span>
-			<span class="hstrut" />
-		      </a>
-		    </li>
-		  </py:otherwise>
-
-		</py:choose>
-	      </py:if>
-	    </py:for>
-
-	    <py:if test="os.status_line != ''">
-	      <li class="statusline">
-		<div class="padded">
-		  ${os.status_line} ${prev_next_links ()}
-		</div>
-	      </li>
-	    </py:if>
-
-	  </ul>
-	</div>
-      </div> <!--! body -->
-	</div>
-  </div>
-  ${site_footer ()}
-
-
-  </body>
+    <div class="container">
+        <div id="content" class="page_content" i18n:comment="On the page of results of a search.">
+
+            <h1><span class="icon icon_${os.title_icon}" />${os.title}</h1>
+
+            <div class="body">
+                <ul class="results">
+                    <p>Sort By:</p>
+                    <div class="sort-dropdown">
+                        <button class="sort-dropdown-toggle" aria-haspopup="true" aria-expanded="false">
+                            <span>${'Title' if os.sort_order == 'alpha' else os.sort_order.replace('_', ' ').title()}</span><span>▼</span>
+                        </button>
+                        <div class="sort-dropdown-menu" role="menu">
+                            <py:for each="x in os.sort_orders">
+                                <py:if test="os.sort_order != x">
+                                    <a href="?sort_order=${x}" role="menuitem">${'Title' if x == 'alpha' else x.replace('_', ' ').title()}</a>
```

Comment:
you need the path in the link here.

Reply:
OK I figured out that it just breaks my test method. (using base element). Nonetheless previous links had path; will need to test _everything_ to make sure there are no places where this code answers at a different url.

Reply:
I updated the links to `os.url_carry(sort_order=order)` which seems to match what was being used before. Let me know if that is better on your end.

Reply:
url_carry is new to me! will try it.